frontend/src/app/Settings/StaffSettings.jsx

```diff
@@ -0,0 +1,291 @@
+import React, { useState } from "react";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import Button from "../commonComponents/Button";
+import { v4 as uuidv4 } from "uuid";
+// import useUniqueId from "../commonComponents/useUniqueIds";  // todo: switch to using
+import useCustomForm from "../commonHooks/FormHook";
+
+// todo: sanitize input somewhere central.
+const MAX_FIELD_LEN = 64;
+
+// todo: move somewhere central. These are roles as defined in the database.
+export const ROLE_TYPES = {
+  ADMIN: "ADMIN",
+  TESTING: "TESTING",
+  INTAKE: "INTAKE",
+};
+
+// todo: replace with something localizable
+const roleToString = (roleType) => {
+  return (
+    {
+      [ROLE_TYPES.ADMIN]: "Admin",
+      [ROLE_TYPES.TESTING]: "Testing",
+      [ROLE_TYPES.INTAKE]: "Intake",
+    }[roleType] || ""
+  );
+};
+
+/**
+ * @param orgSettings {object}
+ * @param updateOrgSettings {function}
+ * @returns {JSX.Element}
+ * @constructor
+ */
+const StaffSettings = ({ orgSettings, updateOrgSettings }) => {
+  const [copyStaffSettings, setCopyStaffSettings] = useState(
+    { ...orgSettings.staff } || {}
+  );
+
+  const saveLocalCopyBackTopParent = () => {
+    // we don't want private fields
+    const staff = {};
+    Object.entries(copyStaffSettings).forEach(([key, value]) => {
+      staff[key] = {
+        name: value.name.substring(0, MAX_FIELD_LEN),
+        email: value.email.substring(0, MAX_FIELD_LEN),
+        role: value.role.substring(0, 50), // todo: restrict to valid strings
+      };
+    });
+    updateOrgSettings({ ...orgSettings, staff });
+  };
+
+  const onStaffAddNew = () => {
+    // Create a new entry in the copy settings
+    // we save the entry out of the copy and into actual settings on save event
+    let newStaffId = `staff-${uuidv4()}`;
+
+    setCopyStaffSettings({
+      ...copyStaffSettings,
+      [newStaffId]: {
+        name: "",
+        email: "",
+        role: ROLE_TYPES.INTAKE,
+        _editable: true,
+      },
+    });
+  };
+
+  const onStaffEditStart = (staffkey) => {
+    // This approach to changing _editable forces the entry to rerender
+    const staffentry = copyStaffSettings[staffkey];
+    setCopyStaffSettings({
+      ...copyStaffSettings,
+      [staffkey]: { ...staffentry, _editable: true },
+    });
+  };
+
+  const onStaffUpdate = ({ staffkey, staffentry }) => {
+    copyStaffSettings[staffkey] = { ...staffentry };
+    setCopyStaffSettings(copyStaffSettings);
+  };
+
+  const onStaffSave = ({ staffkey, staffentry }) => {
+    setCopyStaffSettings({
+      ...copyStaffSettings,
+      [staffkey]: { ...staffentry, _editable: false },
+    });
+    saveLocalCopyBackTopParent();
+  };
+
+  const onStaffRemove = (staffkey) => {
+    // remove entry from ui copy
+    delete copyStaffSettings[staffkey];
```

Comment:
same here. I think react best practices encourages copying state into another variable, making the changes in the copied variable, and then updating state.